src/jsonpatcherproxy.js

```diff
@@ -129,8 +138,8 @@ const JSONPatcherProxy = (function() {
     } else {
       if (isTreeAnArray && !Number.isInteger(+key.toString())) {
         /* array props (as opposed to indices) don't emit any patches, to avoid needless `length` patches */
-        if(key != 'length') {
-          console.warn('JSONPatcherProxy noticed a non-integer prop was set for an array. This will not emit a patch');
+        if(key != 'length' && !warnedAboutNonIntegrerArrayProp) {
```

Comment:
Could you move `const isTreeAnArray = Array.isArray(tree);` from l.118 to l.96. Then here re-use flag set at l.106, so you will not have to re-calculate the condition in l.139.